test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ArrayModelBinderTest.cs

```diff
@@ -151,7 +151,7 @@ private static IModelBinder CreateIntBinder()
                     var model = value.ConvertTo(mbc.ModelType);
                     return ModelBindingResult.Success(model);
                 }
-                return null;
+                return ModelBindingResult.Failed();
```

Comment:
I think this is a little semantically clearer. In general this code is a holdover from when you had to return an `MBR`. In practice neither product nor user code really needs to call `MBR.Failed()`.

Let's start a side discussion about removing `MBR.Failed`.